flink-runtime/src/main/java/org/apache/flink/runtime/state/SnapshotDirectory.java

```diff
@@ -173,7 +173,7 @@ public static SnapshotDirectory permanent(@Nonnull Path directory) throws IOExce
 	private static class TemporarySnapshotDirectory extends SnapshotDirectory {
 
 		TemporarySnapshotDirectory(@Nonnull Path directory) throws IOException {
```

Comment:
Let's only accept a `File` here. That way we know that it needs to be a local file.